### DISCUSSION TITLE
Question regarding GitFlow Yaml Configuration for unique versions for tags, release branch, and develop

### DISCUSSION THREAD

**Discussion by craigbennerAcentra** [GitTools/GitVersion#4238] [Q&A] [answered]
We are looking to utilize gitversion in our CI pipelines to automatically calculate and build the code with versions that are unique based on GitFlow.   We are tagging our production releases with "tags/{major.minor.patch}.  

main
   - tags/0.0.1     --- version==0.0.1     (last release)
release
    - release/0.0.2     --- version==0.0.2    (next release)
develop                   --- version==0.0.3    (current development) <--- this is where I'm struggling to get code configured

features -- would inherit develop's version
hotfix -- would inherit whatever tag it is branched from
support -- would use the branch name like tags and releases

Hopefully this image uploads, as it gives a visual to the sequence diagram that describe the above.

![image](https://github.com/user-attachments/assets/9d1cb1aa-f67e-463d-a90e-eb4f20051f7c)

Can anyone help figure out the correct configuration for what is described above?

thanks,
Craig

**Comment by HHobeck** [comment, marked as answer]
First of all, I would recommend you to take a look into the configuration document and the `GitFlow` examples which are available at:
- [Configuration](https://gitversion.net/docs/reference/configuration)
- [GitFlow Examples](https://gitversion.net/docs/learn/branching-strategies/gitflow/examples)

Seeing your sequence diagram, I think you can use the `workflow: GitFlow/v1` template. You just need to adjust the following:
- the tag-prefix to `tags\/`
- the increment of the `devleop` branch from `Minor` to `Patch`
- the label of the `devleop` branch from `alpha` to `SNAPSHOT`
- the label of the `release` branch from `beta` to `SNAPSHOT`
- the deployment mode (mode) from `ManualDeployment` to `ContinuousDelivery`

Happy Branching!

**Comment by craigbennerAcentra** [comment]
Thank you for your reply.  I have read the configuration multiple times, but for some reason I kept on "ignoring" the `workflow: GitFlow/v1`

Once I added that to my current config it started working :-D

thanks again!
Craig
